test/plugins/validation/oas3/security-definitions-ibm.js

```diff
@@ -0,0 +1,235 @@
+const expect = require('expect');
+const {
+  validate
+} = require('../../../../src/plugins/validation/oas3/semantic-validators/security-definitions-ibm');
+
+describe('it should have a type of `apiKey`,`http`,`oauth2`, `openIdConnect`', function() {
+  it('should have a type', function() {
+    const spec = {
+      components: {
+        securitySchemes: {
+          SecuritySchemeModel: {
+            scheme: 'basic'
+          }
+        }
+      }
+    };
+
+    const res = validate({ resolvedSpec: spec });
+    expect(res.errors.length).toEqual(1);
+    expect(res.warnings.length).toEqual(0);
+    expect(res.errors[0].message).toEqual(
+      "'type' must be defined as a string and is required for path: securitySchemes.SecuritySchemeModel"
+    );
+  });
+  it('type can only be `apiKey`, `http`, `oauth2`, `openIdConnect`', function() {
+    const spec = {
+      components: {
+        securitySchemes: {
+          SecuritySchemeModel: {
+            type: 'wrong type',
+            scheme: 'basic',
+            descriptions: 'example text'
+          }
+        }
+      }
+    };
+
+    const res = validate({ resolvedSpec: spec });
+    expect(res.errors.length).toEqual(1);
+    expect(res.warnings.length).toEqual(0);
+    expect(res.errors[0].message).toEqual(
+      '`type` must have one of the following types: `apiKey`, `oauth2`, `http`, `openIdConnect`'
+    );
+  });
+});
+
+describe('if the type is `apiKey` then it should have `query`, `header` or `cookie` as well as `name`', function() {
```

Comment:
I see what you're doing here, but the `describe` text should provide more of a category. Something like `api key` for rules pertaining to the `apiKey` type would be sufficient. Then the `it` blocks describe what the code should and shouldn't do